src/views/common/TimeScrollView/YAxisTicks.tsx

```diff
@@ -32,27 +32,65 @@ const range = (min: number, max: number, step: number, base: number) => {
         .filter(x => x > min)
 }
 
+/**
+ * Find a set of appropriately-scaled grid lines for the Y axis of a plot.
+ * We assume that we have data over some range, and want to break the Y axis into ticks
+ * that will generate enough grid lines to be useful, but not so many that they will make
+ * it hard to read the plot or otherwise obscure the data.
+ * 
+ * Grid lines are considered valid if each line represents a step of 1, 2, or 5 base units,
+ * where the base unit is some power of 10 of the data displayed in the chart. Conceptually,
+ * if we have measurements from (e.g.) 123 ml to 187 ml, we will first see if 1-ml grid lines work;
+ * this is probably too many, so we'll then try having each grid line/y-axis tick indicate 2 ml;
+ * then 5 ml; then 10 ml, 20 ml, etc. until we eventually find an acceptable size.
+ * 
+ * @param minGridLines Minimum number of grid lines that should be present on the plot.
+ * @param maxGridLines Maximum number of grid lines before the plot appears too busy.
+ * @param range The range of the values displayed in the plot, normalized to ensure it's got 3 digits
+ * before the decimal point.
+ * @returns {step: number, scale: number} a dictionary indicating the scale (power of 10) of the chosen
+ * grid lines and the size of each step (1, 2, or 5) at that order of magnitude. The distance between
+ * any two grid line ticks would then be step * 10 ** scale.
+ */
 const fitGridLines = (minGridLines: number, maxGridLines: number, range: number): {step: number, scale: number} => {
     let scale = 0
     while (true) {
-        const steps = [1, 2, 5]
-        const realizedScale = Math.pow(10, scale)
+        const steps = [1, 2, 5]  // candidate step sizes are 1, 2, 5, (10, 20, 50, 100, 200, ...)
+        const realizedScale = Math.pow(10, scale)   // realizedScale = the base order of magnitude represented by `scale`.
         const results: {step: number, scale: number}[] = []
-        for (let s of steps) {
+        for (let s of steps) {  // for each of the 3 candidate step sizes at this order of magnitude
+            // determine how many steps of [1 | 2 | 5] * 10^scale would be needed to exhaust the range.
+            // (technically on the plot this will be the number of spaces between the grid lines, thus the hard </>).
             const fit = range/(s * realizedScale)
             if (fit > minGridLines && fit < maxGridLines) {
```

Comment:
Probably should use >= and <= here.

Reply:
I think I had some reason for choosing strict equality, probably related to the logic that `fit` (= range over stepsize) determines the number of inter-grid-line intervals, so you wind up having to round it off... but ultimately, I think it doesn't really make a difference because both fit and `minGridLines`/`maxGridLines` (= window `pixelHeight` over the max/min number of pixels between lines) are probably non-integral anyway, so the chances of actual equality are pretty low.

Reply:
won't make much of a difference